File: go/no1679/1679.go
package no1679

func maxOperations(nums []int, k int) int {
	hashMap := make(map[int]int)
	res := 0
	for _, num := range nums {
		val := hashMap[num]
		if val > 0 {
			res += 1
			hashMap[num] -= 1
		} else {
			_, ok := hashMap[k-num]
			if ok {
				hashMap[k-num] += 1
			} else {
				hashMap[k-num] = 1
			}
		}
	}
	return res
}

File: js/source/1679.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maxOperations = function (nums, k) {
  const map = new Map();
  let res = 0;
  nums.forEach((num) => {
    const val = map.get(num);
    if (val > 0) {
      map.set(num, val - 1);
      res += 1;
    } else {
      if (map.has(k - num)) {
        map.set(k - num, map.get(k - num) + 1);
      } else {
        map.set(k - num, 1);
      }
    }
  });
  return res;
};

module.exports = maxOperations;
